src/UmbracoFileSystemProviders.Azure/AzureFileSystem.cs

```diff
@@ -783,11 +784,41 @@ private CloudBlockBlob GetBlockBlobReference(string path)
 
             string blobPath = this.FixPath(path);
 
-            // Only make the request if there is an actual path. See issue 8.
-            // https://github.com/JimBobSquarePants/UmbracoFileSystemProviders.Azure/issues/8
-            return !string.IsNullOrWhiteSpace(path)
-                ? this.cloudBlobContainer.GetBlockBlobReference(blobPath)
-                : null;
+            try
+            {
+                var blobReference = this.cloudBlobContainer.GetBlobReferenceFromServer(blobPath);
+                // Only make the request if there is an actual path. See issue 8.
+                // https://github.com/JimBobSquarePants/UmbracoFileSystemProviders.Azure/issues/8
+                if (blobReference.BlobType == BlobType.BlockBlob && !string.IsNullOrWhiteSpace(path))
+                {
+                    return blobReference as CloudBlockBlob;
+                }
+                else
+                {
+                    Current.Logger.Error<AzureBlobFileSystem>(
+                        $"A media item '{path}' was requested but it's blob type was {blobReference.BlobType} when it should be BlockBlob");
+                    return null;
+                }
+            }
+            catch (StorageException ex) when (ex.RequestInformation.ErrorCode == BlobErrorCodeStrings.BlobNotFound)
```

Comment:
Why wouldn't the blob exist yet? Is this when creating new blobs?

Reply:
It might be adding a blob, this method is used for all actions but in order to add a blob, we need to get a BlockBlobReference even though the reference isn't from the server. The code is currently written assuming that all blobs are blocks so this is only a failsafe. This https://stackoverflow.com/questions/18053679/what-is-the-difference-between-getblobreference-and-getblobreferencefromserver helps a little